src/Admin/SiteHealth.php

```diff
@@ -382,12 +380,40 @@ public function page_cache() {
 		];
 	}
 
+	/**
+	 * Get page caching result from cache.
+	 *
+	 * @param bool $use_previous_result Whether to use previous result or not.
+	 *
+	 * @return bool|mixed|WP_Error
+	 */
+	public function has_page_caching( $use_previous_result = false ) {
+
+		if ( $use_previous_result ) {
+			$has_page_caching = get_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY );
+			if ( is_wp_error( $has_page_caching ) ) {
+				return $has_page_caching; 
+			} elseif ( $has_page_caching ) {
+				return ( 'yes' === $has_page_caching );
+			}
+		}
+
+		$has_page_caching = $this->check_for_page_caching();
+		if ( is_wp_error( $has_page_caching ) ) {
+			set_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY, $has_page_caching, HOUR_IN_SECONDS );
```

Comment:
Thinking about this some more, I think we can can increase this to a daily expiration:

```suggestion
			set_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY, $has_page_caching, DAY_IN_SECONDS );
```